### PR TITLE
#166272358 Read History

### PR DESCRIPTION
**Pull request by EMacco**
#### What does this PR do?
Store and fetch read history for a logged in user.

#### Description of Task to be completed?
- Write tests
- Create middleware to authenticate user
- Add article to read history of a user when the user fetches a single article
- Create route to fetch history for a user `GET /users/history`
- Document swagger

#### How should this be manually tested?
- npm install 
- Login `POST /auth/login`
- Call endpoint to read a particular article `GET /articles/:slug`
- Fetch your history `GET /users/history`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#166272358](https://www.pivotaltracker.com/story/show/166272358)

#### Screenshots (if appropriate)
Screenshot 2019-06-24 at 11 59 23 AM

#### Questions:
N/A

### CODE REVIEW

**Review by adex001 — DISMISSED** [review]
Nice job..

**Review by emmygozi — DISMISSED** [review]
Looks good implementing feedback and syncing with team members

File: server/helpers/Token.js [hunk 1]
```diff
@@ -76,5 +76,38 @@ class Token {
...
+  /**
+   *
+   *
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @param {*} next
+   * @return {null} Returns nothing
+   */
+  static async authenticate(req, res, next) {
```
Comment (fidelisojeah) [inline server/helpers/Token.js]: Why is this here?

Is it any part of your ticket?

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: server/helpers/Token.js (hunk 1 above)
Comment (EMacco) [reply server/helpers/Token.js]: Yes. 
My ticket requires that I authenticate a user if the user is signed in so that I will be able to log the article into the user's read history

If I used the authorize middleware, it will not allow a logged out user to access the route, so I created the authentication middleware so that it identifies the user is he is logged in, but ignores if the user is not.

**Review by fidelisojeah — APPROVED** [review]

**Review by emmygozi — APPROVED** [review]

**Review by adex001 — APPROVED** [review]
